### DISCUSSION TITLE
请教一下， 动态路由表是如何生成的？

### DISCUSSION THREAD

**Discussion by joisun** [arco-design/arco-design-pro-vue#328] [Q&A] [answered]
我想知道动态路由表是如何生成的，  以往我们不是会通过 router.addRoutes 这种方式， 将从服务器拉取的路由表， 动态生成吗。 在 arco pro vue 中， 这部分逻辑是如何实现的？ 我有注意到 在 https://github.com/arco-design/arco-design-pro-vue/blob/7350c03977abc9e258de13eaf08e8461c818292a/arco-design-pro-vite/src/router/guard/permission.ts#L27C13-L27C29
这里， 有合并拉取的路由表， 但是这个是menu, 那么router 呢？ router 中 routes 的 component 在哪里配置的呢？  多谢

**Comment by shangwenhe** [comment, marked as answer]
你需要理解一下，将路由与菜单的两者分开。不要混淆。

真正的路由还是前在前端。服务端返回的都是菜单以及地址，它不需要配置 component。

另：不太了解你的场景。如果你是低代码场景，你的 component 也是后端生成。那你需要有一个对应的动态模块加载器。去加载你的 component。

**Reply by joisun** [reply]
非常感谢你的回复， 这部分内容我已经明白了， 确实如你所说。
